runtime/parachains/src/initializer.rs

```diff
@@ -0,0 +1,148 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! This module is responsible for maintaining a consistent initialization order for all other
+//! parachains modules. It's also responsible for finalization and session change notifications.
+//!
+//! This module can throw fatal errors if session-change notifications are received after initialization.
+
+use sp_std::prelude::*;
+use frame_support::weights::Weight;
+use primitives::{
+	parachain::{ValidatorId},
+};
+use frame_support::{
+	decl_storage, decl_module, decl_error,
+};
+use crate::{configuration, paras};
+
+pub trait Trait: system::Trait + configuration::Trait + paras::Trait { }
+
+decl_storage! {
+	trait Store for Module<T: Trait> as Initializer {
+		/// Whether the parachains modules have been initialized within this block.
+		/// Should never hit the trie.
```

Comment:
Since it's an option, the `None` variant does not result in any storage being committed to the state trie, and we clear out any `Some` variant in `on_finalize`, preventing that from hitting the trie either.